Test/ServerTelemetryChannel.Test/TestFramework/Shared/StubTransmission.cs

```diff
@@ -35,5 +41,58 @@ public override Task<HttpWebResponseWrapper> SendAsync()
         {
             return TaskEx.Run(this.OnSend);
         }
+
+        public override Tuple<Transmission, Transmission> Split(Func<int, int> calculateLength)
+        {
+            var ret = base.Split(calculateLength);
+
+            if (ret.Item2 == null)
+            {
+                return Tuple.Create((Transmission)this, (Transmission) null);
+            }
+
+            return Tuple.Create((Transmission)this.Convert(ret.Item1), (Transmission)this.Convert(ret.Item2));
+        }
+
+        private StubTransmission Convert(Transmission transmission)
+        {
+            if (transmission != null)
+            {
+                if (transmission.TelemetryItems == null)
+                {
+                    transmission = new StubTransmission(transmission.Content)
+                    {
+                        OnSave = this.OnSave,
+                        OnSend = this.OnSend,
```

Comment:
Adding the extra properties still didn't allow for the equality check to pass. Instead the stub uses the same functionality as them real method: the exact object is returned if no split is necessary. The Ids changing on actual splits makes sense (they are new transmissions). Ids are only used for logging purposes from what I can find, so there should be no functional impact.